libs/counterfactuals/src/lib/CounterfactualChartLegend.tsx

```diff
@@ -52,23 +59,26 @@ export class CounterfactualChartLegend extends React.PureComponent<ICounterfactu
     const classNames = counterfactualChartStyles();
     return (
       <Stack className={classNames.legendAndText}>
-        <ComboBox
-          id={"CounterfactualSelectedDatapoint"}
-          className={classNames.legendLabel}
-          label={localization.Counterfactuals.selectedDatapoint}
-          onChange={this.selectPointFromDropdown}
-          options={this.getDataOptions()}
-          selectedKey={`${this.props.selectedPointsIndexes[0]}`}
-          ariaLabel={"datapoint picker"}
-          useComboBoxAsMenuWidth
-          styles={FluentUIStyles.smallDropdownStyle}
-        />
+        {this.displayDatapointDropbox() && (
+          <ComboBox
+            id={"CounterfactualSelectedDatapoint"}
+            className={classNames.legendLabel}
+            label={localization.Counterfactuals.selectedDatapoint}
+            onChange={this.selectPointFromDropdown}
+            options={this.getDataOptions()}
+            selectedKey={`${this.props.selectedPointsIndexes[0]}`}
+            ariaLabel={"datapoint picker"}
+            useComboBoxAsMenuWidth
+            styles={FluentUIStyles.smallDropdownStyle}
+            disabled={this.props.isCounterfactualsDataLoading}
+          />
+        )}
         <div className={classNames.legendLabel}>
```

Comment:
While we're already here, do we want to convert these `<b>` and `<div>` tags with FluentUI ones like `Label` and `Stack`?

Reply:
I observed Stack has to be used with Stack.Items else will result in overflow of components inside Stack. It would be future engineering task to remove these and test thoroughly with reflow cases.

Reply:
Fair. Since you're not touching these I don't want to burden you with more work 🙂